hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sDatabaseTable.java

```diff
@@ -82,9 +83,28 @@ private static String schemaName(Row row) {
     }
   }
 
-  private static DataSource dataSource(Row row) {
-    // TODO fetch username/password from Secret
-    return JdbcSchema.dataSource(row.URL, row.DRIVER, "nouser", "nopass");
+  private static DataSource dataSource(Row row, Properties connectionProperties) {
+    String user = "nouser";
+    String pass = "nopass";
+    StringJoiner joiner = new StringJoiner(";");
+    for (String key : connectionProperties.stringPropertyNames()) {
+      if ("user".equals(key)) {
+        user = connectionProperties.getProperty(key);
+      } else if ("password".equals(key)) {
+        pass = connectionProperties.getProperty(key);
+      } else {
+        String value = connectionProperties.getProperty(key);
+        joiner.add(key + "=" + value);
+      }
+    }
+    String joinedUrl = row.URL;
+    // Handles case where there are no properties already in the URL
+    if (row.URL.endsWith("//")) {
+      joinedUrl = joinedUrl + joiner;
+    } else {
+      joinedUrl = joinedUrl + ";" + joiner;
```

Comment:
The `k1=v1;k2=v2` format seems to be a common convention, but not all drivers support it. We may run into drivers that choke on it. Not sure what to do in that case...

Reply:
All the drivers we have for now do support it, so maybe that's good enough for now? Considering Properties objects are always key/value pairs I'd imagine we'd run into more problems if we just had single values